src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -1100,7 +1134,12 @@ void MacroAssembler::get_vm_result_2(Register metadata_result, Register java_thr
 }
 
 void MacroAssembler::align(int modulus) {
-  while (offset() % modulus != 0) nop();
+  align(modulus, offset());
+}
+
+void MacroAssembler::align(int modulus, int target) {
```

Comment:
It would be nice to document what this extra `align` function does.

Reply:
Good idea. I wrote a comment.